src/h5web/providers/hsds/api.ts

```diff
@@ -35,12 +36,13 @@ export class HsdsApi implements ProviderAPI {
   private datatypes: Record<HDF5Id, HDF5Datatype> = {};
   private values: Record<HDF5Id, HDF5Value> = {};
 
-  constructor(private readonly domain: string) {
+  constructor(username: string, password: string, filepath: string) {
```

Comment:
Need `private readonly` in front of every param.

Reply:
Do I really need to if I only want to keep the config as a member of the class and not the individual parameters ?